arviz/data/inference_data.py

```diff
@@ -177,6 +177,11 @@ def __delattr__(self, group):
     def _groups_all(self):
         return self._groups + self._groups_warmup
 
+    def __iter__(self):
```

Comment:
Do we wan't to expose this with __iter__ or with `.keys`, `.items`, `.values`?

Or how dict does it?

Reply:
In my opinion, `__iter__` will be more useful. Maybe we can add a `.groups` function (like keys) which will be more intuitive . What are your thoughts on this @OriolAbril ?

Reply:
I think dict like will be more flexible. 

What will iter loop over? Group names? Xarray datasets? A tuple of (group name, dataset)? I think all 3 can have their uses, hence my preference for the dict like approach, as I said somewhere, renaming the `keys` to something else, maybe `groups`. __iter __ could do the same as keys as it happens for dicts or we could eventually decide on forcing expliciteness and have __iter __ raise an error explaining the 3 alternatives

Reply:
Ok, I'll do that.

Reply:
Should I add`filter_groups` to the values?

Reply:
I think its probably better to keep the functions as close as possible as dict counterparts, so no extra arguments. 

It is true however that there is no clear way of subsetting an inferencedata on a group basis other than deleting groups. Should there be a way to do that? or doing `for ... in idata... if <condition>` is enough?

Reply:
I think adding extra arguments with default values should not be much of a problem as we can still use `groups()` similar to `keys()`. I think this will be a cleaner way to subset the idata groups.

Reply:
I have a slight preference towards having a subsetting function and the looping ones separated because is that setting, one can do any of `idata.<subsetting>.items()`, `idata.<subsetting>` or `idata.items()` whereas adding the subsetting inside `items()` and `groups` would not allow pure subsetting yet. But again, it's a slight preference, let's see what other people thinks cc @ahartikainen @aloctavodia @AlexAndorra